rust/arrow/src/compute/cast_kernels.rs

```diff
@@ -0,0 +1,543 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
can we have a `kernels` directory under `compute` and rename this to `cast.rs`?

Reply:
I guess this is to be consistent with cpp?  In that case we should move all the existing kernels too (boolean, comparison).

Reply:
Yes. I think we should do that too.